src/components/sparks/Sparks.tsx

```diff
@@ -0,0 +1,173 @@
+   **/
+  duration?: number;
+
+  /**
+   * Optional number. Delay of the animation.
+   * This value is used at the beginning of each iteration.
+   * @default 0
+   * @example <Sparks delay={0}/>
+   **/
+  delay?: number;
+
+  /**
+   * Optional number. Number of iterations to play the full sequence: entry, exit, delay.
+   * @default 3
+   * @example
+   * <Sparks iterationCount={3}/>
+   * <Sparks iterationCount={Infinity}/>
+   **/
+  iterationCount?: number;
+}
+
+const Sparks = React.forwardRef<HTMLDivElement, SparksPropsType>(
+  (
+    {
+      children,
+      className,
+      style,
+      shape = 'spark',
+      variant = 'l',
+      active = false,
+      duration = 6000,
+      delay = 0,
+      iterationCount = 3,
+    }: SparksPropsType,
+    ref
+  ) => {
+    const iteration = React.useRef(0);
+    const animationConfig = shapeAnimationMap[shape];
+    const {register, phase, setPhase} = useAnimation(animationConfig);
+    const [timeoutDelay, setTimeoutDelay] = React.useState<number | null>(null);
+
+    const restartTimeout = useTimeout(() => {
+      if (!active) {
+        return;
+      }
+
+      if (phase === 'entry') {
+        setPhase('exit');
+      } else if (iteration.current < iterationCount) {
+        setPhase('entry');
+      }
+    }, timeoutDelay);
+
+    React.useEffect(() => {
+      if (active) {
+        setPhase('initial');
+      } else {
+        setPhase('exit');
+      }
+    }, [active, setPhase]);
+
+    React.useEffect(() => {
+      // eslint-disable-next-line no-console
+      //console.log(`Play animation %c${phase}`, 'background: #000; color: #fff');
+
+      if (phase === 'initial' && active) {
+        iteration.current = 0;
+        restartTimeout();
+        setTimeoutDelay(delay);
+      }
+
+      if (phase === 'entry' && active) {
+        iteration.current++;
+
+        if (duration !== Infinity) {
+          restartTimeout();
+          setTimeoutDelay(duration);
+        }
+      }
+
+      if (phase === 'finished') {
+        restartTimeout();
+        if (delay !== Infinity) {
+          setTimeoutDelay(delay);
+        }
+      }
+    }, [phase, delay, active, duration, iterationCount, restartTimeout]);
+
+    const shapeColor = shapeColorMap[shape];
+
+    return (
+      <div
+        className={cx('sg-sparks', className, {
+          'sg-sparks--s': variant === 's',
+        })}
+        style={style}
+      >
+        {children}
+        <div className="sg-sparks__container" aria-hidden ref={ref}>
+          {variants[variant].map(
+            ({style, colorIndex, animation, ...particle}, index) => (
+              <Particle
+                key={index}
+                shape={shape}
+                style={{...style, color: shapeColor[colorIndex]}}
+                {...particle}
+                {...register(animation)}
+              />
+            )
+          )}
+        </div>
+      </div>
+    );
+  }
+);
+
+Sparks.displayName = 'Sparks';
```

Comment:
I think we can remove it due @babel/plugin-transform-react-display-name is doing that for us. We have it as a part of [@babel/preset-react](https://babeljs.io/docs/babel-preset-react)

Reply:
True